docs/src/references/tedge-config-dir.md

```diff
@@ -0,0 +1,73 @@
+# `tedge --config-dir`
+The `tedge` CLI `config-dir` option can be used to store the thin-edge
+configuration files in a custom directory.
+
+## Create all the directories
+Create all the directories that are needed by the thin-edge for its operations in a custom path as below.
+
+```shell
+
+tedge --config-dir path/to/config/dir --init
+
+```
+All the directories will be created in the `path/to/config/dir` directory.
+
+Create the directories and files that are required by the `tedge_mapper`
+and `tedge_agent` in a custom directory as below.
+
+```shell
+
+tedge_mapper --config-dir path/to/config/dir --init c8y
+tedge_agent --config-dir path/to/config/dir --init
+
+```
+
+## Manage the configuration parameters
+
+The configuration parameters can be set/unset/list in a config file that is present
+in a custom directory.
+
+For example, the config parameter can be set as below
+
+```shell
+
+tedge --config-dir path/to/config/dir config set c8y.url thin-edge.io
+
+```
+
+## Manage the certificate
+
+This involves create/remove/upload the certificate.
+
+For example, the certificate can be created as below
+
+
+```shell
+
+tedge --config-dir path/to/config/dir cert create --device-id thinedge
+
+```
+
+Now the certificate will be created in the `path/to/config/dir` directory.
+
+
+## Tedge connect with `config-dir`
+
+This is a two-step process
+
+### Step 1: Update the `mosquitto.conf`
+
+Since the `c8y/az bridge config` files will be created in a directory given through
+`config-dir`, this path must be updated in the `mosquitto.conf` file. So, this has to be updated manually.
```

Comment:
```suggestion
Since the bridge configuration files for Cumulocity IoT or Azure IoT Hub will be created in a directory given through `--config-dir`,
the path to the bridge configuration files must be found by `mosquitto`.
So, this line has to be added to your `mosquitto.conf` file manually.
```